meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-atf/imx-atf/0001-plat-imx8m-obtain-boot-set-from-bootrom-even-log.patch

```diff
@@ -0,0 +1,96 @@
+From 58cd2feeb07e496c7884f6655f258eca2a25b7ff Mon Sep 17 00:00:00 2001
+From: Igor Opaniuk <igor.opaniuk@foundries.io>
+Date: Fri, 18 Mar 2022 14:52:22 -0300
+Subject: [PATCH] plat: imx8m: obtain boot set from bootrom even log
+
+Both i.MX8MP and i.MX8MN don't have a dedicated bit which indicates
+a boot image set booted by BootROMv According to AN12853 [1] i.MX ROMs Log
+Events, it is possible to determine whether fallback event occurred
+by parsing the ROM event log. In case ROM event ID 0x51 is present,
+fallback event did occur.
+
+The code is based on U-Boot implementation [2].
+
+[1]: https://www.nxp.com/webapp/Download?colCode=AN12853
+[2]: https://github.com/u-boot/u-boot/commit/a5ee05cf7180b411ffdf148ca8cb220c029f2e19
+Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>
+---
+ plat/imx/imx8m/gpc_common.c | 55 +++++++++++++++++++++++++++++++++++++
```

Comment:
Why changing atf? Is the rom log private to secure execution?

Reply:
technically we can do that in U-Boot, I've just cross-checked that we can obtain BootROM log:

```
u-boot=> md 0x9e0  # obtain BootROM log event buffer address (0x009112f8)
000009e0: 009112f8 00000000 009121a0 00000000  .........!......
000009f0: 00000000 00000000 00000000 00000000  ................
00000a00: a9be7bfd f0004861 91388021 910003fd  .{..aH..!.8.....
00000a10: f9400020 f9000fa0 d2800000 f9400fa2   .@...........@.
00000a20: f9400021 ca010041 5280a0c0 72a00080  !.@.A......R...r
00000a30: b4000041 940053e8 a8c27bfd d65f03c0  A....S...{...._.
00000a40: a9bd7bfd 910003fd a90153f3 aa0003f4  .{.......S......
00000a50: f0004873 91388260 f9400001 f90017a1  sH..`.8...@.....
00000a60: d2800001 52800660 39008fa0 52801980  ....`..R...9...R
00000a70: b90027a0 94002c7c b90027a0 b94027a0  .'..|,...'...'@.
00000a80: 52000c00 53001c00 39008fa0 b94027a0  ...R...S...9.'@.
00000a90: 7103301f 540000e0 39408fa0 b94027a1  .0.q...T..@9.'@.
00000aa0: 53001c00 4a210000 53001c00 b9000280  ...S..!J...S....
00000ab0: 91388273 f94017a1 f9400260 ca000020  s.8...@.`.@. ...
00000ac0: b4000040 940053c4 a94153f3 a8c37bfd  @....S...SA..{..
00000ad0: d65f03c0 a9b87bfd 910003fd a90153f3  .._..{.......S..
u-boot=> md 009112f8 # show events from event buffer
009112f8: e0000000 22000000 31000000 40000000  ......."...1...@
00911308: 61000000 82000000 00000ed9 83000000  ...a............
00911318: 80000000 00000ee2 81000000 00009c37  ............7...
00911328: 50000000 90000000 00000000 91000000  ...P............
00911338: 00009ebc a1e00001 51000000 90000000  ...........Q....
00911348: 00000000 91000000 0000a155 90000000  ........U.......
00911358: 00000008 91000000 0000d357 a0000000  ........W.......
00911368: 00cf1133 0000d466 c0000000 00920000  3...f...........
00911378: 0000e758 00000000 00000000 00000000  X...............
00911388: 00000000 00000000 00000000 00000000  ................
00911398: 00000000 00000000 00000000 00000000  ................
009113a8: 00000000 00000000 00000000 00000000  ................
009113b8: 00000000 00000000 00000000 00000000  ................
009113c8: 00000000 00000000 00000000 00000000  ................
009113d8: 00000000 00000000 00000000 00000000  ................
009113e8: 00000000 00000000 00000000 00000000  ................

```

But as we are already using TF-A for obtaining secondary boot bit on i.MX8 MM (IMX_SIP_SRC call), why not to use the same interface for i.MX8MN and i.MX8MN?

Reply:
Sure, if you think this is something we can upstream to tf-a, then we should be fine, otherwise u-boot would probably be the way to land the updates.